MdeModulePkg/Include/Library/MemoryBucketLib.h

```diff
@@ -0,0 +1,130 @@
+/** @file

```

Comment:
It's a little unclear who is expected to call each function in this file.

For each function, are they expected to be called by (1) PeiMain (2) the platform (3) others/both?

Thinking from the perspective of a platform integrating this change, what do they need to do with these functions?

Reply:
Currently everything is only expected to be called by PEI memory services unless a platform wants to touch the memory allocation itself.  As I mentioned above I could consolidate MemoryBucketLib with MemoryServices to reduce confusion.